classes/data-stores/ActionScheduler_DBStore.php

```diff
@@ -1025,6 +1028,36 @@ protected function claim_actions( $claim_id, $limit, ?DateTime $before_date = nu
 		return (int) $rows_affected;
 	}
 
+	/**
+	 * Determines whether the database supports using SKIP LOCKED.
+	 *
+	 * SKIP_LOCKED support was added to MariaDB in 10.6.0 and to MySQL in 8.0.1
+	 *
+	 * @return bool
+	 */
+	private function db_supports_skip_locked() {
```

Comment:
Should maybe cache this check so it doesn't have to confirm the version support before every claim. Can do local runtime cache, but I'd prolly just do object cache with some relatively short TTL like 5 minutes even.

Reply:
@WPprodigy, I don't think that is necessary.  My `mysqli::$server_info` is a read-only property of the underlying `mysqli` object.  I can't find documentation explicitly saying it, but I believe it's set when the connection is made or at the very least would be saved within the connected `mysqli` object property between calls. https://www.php.net/manual/en/class.mysqli.php